packages/amplication-server/src/core/entity/entity.service.ts

```diff
@@ -609,6 +614,36 @@ export class EntityService {
     return true;
   }
 
+  async updateFieldDataTypeIdByRelatedEntity(
+    field: EntityField,
+    relatedEntityId: string
+  ): Promise<EntityField> {
+    const relatedIdField = await this.prisma.entityField.findFirst({
+      where: {
+        dataType: EnumDataType.Id,
+        entityVersion: {
+          entityId: relatedEntityId,
+        },
+      },
+    });
+
+    const idProp = relatedIdField.properties as unknown as {
+      [key: string]: JsonValue;
+    };
+
+    const idType = idProp["idType"];
```

Comment:
```suggestion
    const idType: types.Id["idType" ]= idProp["idType"];
```

Reply:
@abrl91  Did you mean something like that: 
<img width="576" alt="image" src="https://github.com/amplication/amplication/assets/97830649/0697a4be-d08f-4b5e-8006-890abdc77bba">